### PR TITLE
[FLINK-26035][build][planner] Add table-planner-loader-helper module

### PR DESCRIPTION
**Pull request by zentol**
The flink-table-planner currently creates 2 artifacts. 1 jar containing the planner and various dependencies for the cases where the planner is used directly, and another jar that additionally bundles scala for cases where the loader is used.

The latter artifact is purely an intermediate build artifact, and as such we usually wouldn't want to publish it. This is particularly important because this jar doesn't have a correct NOTICE, and having different NOTICE files for different artifacts is surprisingly tricky.

We should just rework this into a separate module.

### CODE REVIEW

File: flink-table/flink-table-planner-loader-helper/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-table</artifactId>
+		<version>1.15-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-table-planner-loader-helper</artifactId>
```
Comment (slinkydeveloper) [inline flink-table/flink-table-planner-loader-helper/pom.xml:33]: Perhaps name it loader bundle?

**Review by slinkydeveloper — COMMENTED** [review]
Can you update the README in flink-table?

File: flink-table/flink-table-planner-loader-helper/pom.xml (hunk 1 above)
Comment (zentol) [reply flink-table/flink-table-planner-loader-helper/pom.xml:33]: sure!

File: flink-table/README.md [hunk 2]
```diff
@@ -65,7 +66,8 @@ If you want to use Table API & SQL, check out the [documentation](https://nightl
 
 ### Notes
 
-No module except `flink-table-planner` should depend on `flink-table-runtime` in production classpath, 
+No module except `flink-table-planner` should depend on `flink-table-runtime` in production classpath,
```
Comment (zentol) [inline flink-table/README.md:69]: @slinkydeveloper Have you considered enforcing this via the enforcer plugin, similarly to what we do for [flink-rpc-akka](https://github.com/apache/flink/blob/master/pom.xml#L1730)?

Comment (slinkydeveloper) [reply flink-table/README.md:69]: I didn't knew about that, but yep it probably makes sense to do so

Comment (slinkydeveloper) [reply flink-table/README.md:69]: Is it possible to do so but allowing flink-table-planner for testing purpose?

Comment (slinkydeveloper) [reply flink-table/README.md:69]: Gotcha, opened an issue here to track it https://issues.apache.org/jira/browse/FLINK-26054

**Review by slinkydeveloper — APPROVED** [review]

**Pull request merged**